app.rb

```diff
@@ -0,0 +1,196 @@
+require_relative "./lib/database_connection"
+require_relative "./lib/item_repository"
+require_relative "./lib/order_repository"
+require_relative "./lib/item"
+require_relative "./lib/order"
+require 'date'
+require 'colorize'
+
+class Application
+  def initialize(database_name, io, item_repository, order_repository)
+    DatabaseConnection.connect(database_name)
+    @io = io
+    @item_repository = item_repository
+    @order_repository = order_repository
+  end
+
+  def run
+    print_welcome
+    loop do
+      print_interface
+      user_selection(@io.gets.chomp)
+    end
+  end
+
+  def print_welcome
+    @io.puts "\n"
+    @io.puts "Welcome to the shop management program!".colorize(:blue)
+  end
+
+  def print_interface
+    @io.puts "\n"
+    @io.puts "Please select one of the options:".colorize(:blue)
+    @io.puts "1 = list all shop items"
+    @io.puts "2 = create a new item"
+    @io.puts "3 = list all orders"
+    @io.puts "4 = create a new order"
+    @io.puts "5 = exit program"
+    @io.puts "\n"
+  end
+
+  def user_selection(input)
+    @io.puts "You selected option #{input}.".colorize(:blue)
+    @io.puts "\n"
+    case input
+    when "1"
+      list_items
+    when "2"
+      create_item
+    when "3"
+      list_orders
+    when "4"
+      create_order
+    when "5"
+      exit
+    else 
+      @io.puts "Invalid! Please enter valid option.".colorize(:red)
```

Comment:
these colorized elements are deployed well and will help user error messages jump out.